gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -487,8 +487,10 @@ private static void cleanupStagingDataPerTask(JobState jobState) {
     Map<String, ParallelRunner> parallelRunners = Maps.newHashMap();
     try {
       for (TaskState taskState : jobState.getTaskStates()) {
+        TaskState fatTaskState = new TaskState(taskState);
+        fatTaskState.addAllIfNotExist(jobState);
```

Comment:
This could be slow if the number of tasks is large. Can you just add the properties needed?